.github/workflows/Build-Rock.yaml

```diff
@@ -1,54 +1,65 @@
-name: Build rock
+name: Build Rock
 
 on:
   workflow_call:
     inputs:
+      # build parameters
       oci-archive-name:
-        description: "Final filename of the rock's OCI archive"
-        type: string
-        required: true
-      oci-factory-path:
-        description: "Path, in the OCI Factory, to this rock"
-        type: string
-        required: true
-      rock-name:
-        description: "Name of the rock"
+        description: "Final filename of the Rock OCI archive."
         type: string
         required: true
+        
+      # source parameters
       rock-repo:
-        description: "Public Git repo where to build the rock from"
+        description: "Public Git repo where to build the rock from."
         type: string
         required: true
       rock-repo-commit:
-        description: "Git ref from where to build the rock from"
+        description: "Git ref from where to build the rock from."
         type: string
         required: true
       rockfile-directory:
-        description: "Directory, in 'rock-repo', where to find the rockcraft.yaml file"
+        description: "Directory in repository where to find the rockcraft.yaml file."
         type: string
         required: true
 
+      # optional parameters for multi-arch builds
+      arch-map:
+        description: "JSON string mapping target architecture to runners."
+        type: string
+        default: '{"amd64": ["linux", "X64"], "arm64": ["linux", "ARM64"]}'
+      lpci-fallback:
+        description: 'Enable fallback to Launchpad build when runners for target arch are not available.'
+        type: boolean
+        default: false
+
+
+
 env:
-  ROCKS_CI_FOLDER: ci-rocks
+  ROCK_REPO_DIR: rock-repo  # path where the image repo is cloned to
+  ROCK_CI_FOLDER: ci-rocks  # path of uploaded/downloaded artifacts
+  
+  OCI_FACTORY_REPO: canonical/oci-factory
+  # TODO: update before final merge
+  OCI_FACTORY_BRANCH: ROCKS-1452/oci-factory-refactor-build-rock-to-be-externally-reusable  # required for external workflow calls.
```

Comment:
My initial plan was to have a short follow up PR, where I update OCI_FACTORY_BRANCH to reflect main. But thinking back now, I think think this will make development awkward since we always reference that branch, even for testing. I think we can set OCI_FACTORY_BRANCH in such that we only enforce the default branch if the call is external, which is where we need this mechanism anyhow. 

Passing a ref in a re usable workflow seem to be a [common issue](https://github.com/actions/toolkit/issues/1264). If it is ok with you, I'll add a ticket for this, and address it in a future PR. https://warthogs.atlassian.net/browse/ROCKS-1583

Reply:
I found a useful action which get around this issue, and returns the ref that was used to call the workflow. The only caveat is that it requires a GH token for private repositories.  I think we can add support for that as needed.

https://github.com/canonical/get-workflow-version-action